pkg/node/node.go

```diff
@@ -278,12 +283,22 @@ func createAPIServer(cfg NodeConfig, userKey *baccrypto.UserKey) (*publicapi.Ser
 		return nil, err
 	}
 
+	givenPortNumber := cfg.BacalhauConfig.API.Port
+	if givenPortNumber < minPortNumber {
+		return nil, fmt.Errorf("invalid negative port number: %d", cfg.BacalhauConfig.API.Port)
+	}
+	if givenPortNumber > maxPortNumber {
+		return nil, fmt.Errorf("port number %d exceeds maximum allowed value (65535)", cfg.BacalhauConfig.API.Port)
+	}
+
+	safePortNumber := uint16(givenPortNumber)
+
```

Comment:
nit: might make sense to include these validations in `Validate()` method of `config.Bacalhau` or of `node.NodeConfig`